src/language_concepts/data_types.md

```diff
@@ -1,206 +1,192 @@
-# Data types
+# Data Types
 
-Noir has several varieties of data types: the primitive `Field` type, integer types, `bool`, arrays, structs, and tuple types.
+Every value in Noir is specified with a certain data type, which informs Noir tools how to work with that data when processing Noir programs.
 
-Although each value in a constraint system is fundamentally a field element, we add a layer of abstraction over this;
-each value can be _concealed_ (a private type) or _revealed_ (a public type).
+All values in Noir are fundamentally `Field` elements. For a more approachable developing experience, abstractions are added on top to introduce different data types in Noir.
 
-A **private value** is known only to the Prover, while, a **public value** is known by the Prover and Verifier.
-All public and private values are primitive types, though individual
-fields of a struct may also be marked private or public.
+Noir has two category of data types: primitive types (e.g. `Field`, integer, `bool`) and compound types (e.g. array, tuple, struct). Each value can either be private or public.
```

Comment:
lets change "integer" to "integers" or "signed/unsigned integers". The compound types listed should also be plural - "arrays, tuples, and structs"